esmvaltool/cmorizers/data/formatters/datasets/esdc.py

```diff
@@ -0,0 +1,117 @@
+    website, and downloaded using wget:
+        ```wget -m -nH http://url/to/dataset```
+"""
+import logging
+from pathlib import Path
+
+import xarray as xr
+from esmvalcore.preprocessor import monthly_statistics
+
+from esmvaltool.cmorizers.data import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+
+def fix_cube(short_name, var, cube, cfg):
+    """General fixes for all cubes."""
+    cmor_info = cfg['cmor_table'].get_variable(var['mip'], short_name)
+    utils.fix_coords(cube)
+    cube.convert_units(cmor_info.units)
+    if 'height2m' in cmor_info.dimensions:
+        utils.add_height2m(cube)
+    cube = monthly_statistics(cube, operator="mean")  # Fix frequency
+    # Fix metadata
+    attrs = cfg['attributes']
+    attrs['mip'] = var['mip']
+    utils.fix_var_metadata(cube, cmor_info)
+    utils.set_global_atts(cube, attrs)
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize the dataset."""
+    # 1. find the input data
+    logger.debug("in_dir: '%s'", in_dir)
+    logger.debug("cfg: '%s'", cfg)
+    logger.debug("cfg_user: '%s'", cfg_user)
+
+    if start_date:
+        logger.warning('start_date set to "%s", but will be ignored',
+                       start_date)
+    if end_date:
+        logger.warning('end_date set to "%s", but will be ignored', end_date)
+
+    attributes = cfg['attributes']
+    variables = cfg['variables']
+
+    version = attributes['version']
+
+    filename_pattern = cfg['filename'].format(grid=attributes['grid'],
+                                              chunking=attributes['chunking'],
+                                              version=version)
+
+    def cmorize_cube(path):
+        logger.info('Opening zarr in "%s"', path)
+        try:
+            dataset = xr.open_dataset(path, engine='zarr')
+        except KeyError as exception:
+            # Happens when the zarr folder is missing metadata, e.g. when
+            # it is a zarr array instead of a zarr dataset.
+            logger.info('Could not open zarr dataset "%s": "KeyError: %s"',
+                        path, exception)
+            logger.info('Skipping path')
+            return
+
+        for short_name, var in variables.items():
+            all_attributes = {
+                **attributes,
+                **var
+            }  # add the mip to the other attributes
+            raw_name = var['raw']
+            cube_xr = dataset[raw_name]
+
+            cube_iris = cube_xr.to_iris()
+
+            # 2. apply the necessary fixes
+            cube = fix_cube(short_name, var, cube_iris, cfg)
+
+            # 3. store the data with the correct filename
+            utils.save_variable(cube=cube,
+                                var=short_name,
+                                outdir=out_dir,
+                                attrs=all_attributes)
+
+    local_path = Path(in_dir, f'v{version}')
+    matches = list(local_path.glob(filename_pattern))
```

Comment:
I would not recommend to support the usage of wildcards in the data search, especially because this would only work with local data but not data from cloud stores (not findable with wildcards). We discussed already in some meetings or PRs (I can't recall exactly) that it is preferable to use explicit values for the version, resolution, ... so that we have more control over the CMORizer input.

Reply:
Agree, support for multiple files makes the code awkward for minimal gain. I've cleaned it up.